toolkits/wenet/gigaspeech_data_prep.sh

```diff
@@ -0,0 +1,142 @@
+train_subset=XL
+
+. ./utils/parse_options.sh || exit 1;
+
+filter_by_id () {
+  idlist=$1
+  input=$2
+  output=$3
+  field=1
+  if [ $# -eq 4 ]; then
+    field=$4
+  fi
+  cat $input | perl -se '
+    open(F, "<$idlist") || die "Could not open id-list file $idlist";
+    while(<F>) {
+      @A = split;
+      @A>=1 || die "Invalid id-list file line $_";
+      $seen{$A[0]} = 1;
+    }
+    while(<>) {
+      @A = split;
+      @A > 0 || die "Invalid file line $_";
+      @A >= $field || die "Invalid file line $_";
+      if ($seen{$A[$field-1]}) {
+        print $_;
+      }
+    }' -- -idlist="$idlist" -field="$field" > $output ||\
+  (echo "$0: filter_by_id() error: $input" && exit 1) || exit 1;
+}
+
+subset_data_dir () {
+  utt_list=$1
+  src_dir=$2
+  dest_dir=$3
+  mkdir -p $dest_dir || exit 1;
+  # wav.scp text segments utt2dur
+  filter_by_id $utt_list $src_dir/utt2dur $dest_dir/utt2dur ||\
+    (echo "$0: subset_data_dir() error: $src_dir/utt2dur" && exit 1) || exit 1;
+  filter_by_id $utt_list $src_dir/text $dest_dir/text ||\
+    (echo "$0: subset_data_dir() error: $src_dir/text" && exit 1) || exit 1;
+  filter_by_id $utt_list $src_dir/segments $dest_dir/segments ||\
+    (echo "$0: subset_data_dir() error: $src_dir/segments" && exit 1) || exit 1;
+  awk '{print $2}' $dest_dir/segments | sort | uniq > $dest_dir/reco
+  filter_by_id $dest_dir/reco $src_dir/wav.scp $dest_dir/wav.scp ||\
+    (echo "$0: subset_data_dir() error: $src_dir/wav.scp" && exit 1) || exit 1;
+  rm -f $dest_dir/reco
+}
+
+if [ $# -ne 2 ]; then
+  echo "Usage: $0 [options] <gigaspeech-dataset-dir> <data-dir>"
+  echo " e.g.: $0 --train-subset XL /disk1/audio_data/gigaspeech/ data/"
+  echo ""
+  echo "This script takes the GigaSpeech source directory, and prepares the"
+  echo "WeNet format data directory."
+  echo "  --garbage-utterance-tags <tags>  # Tags for non-speech."
+  echo "  --prefix <prefix>                # Prefix for output data directory."
+  echo "  --punctuation-tags <tags>        # Tags for punctuations."
+  echo "  --stage <stage>                  # Processing stage."
+  echo "  --train-subset <XL|L|M|S|XS>     # Train subset to be created."
+  exit 1
+fi
+
+gigaspeech_dir=$1
+data_dir=$2
+
+declare -A subsets
+subsets=(
+  [XL]="train_xl"
+  [L]="train_l"
+  [M]="train_m"
+  [S]="train_s"
+  [XS]="train_xs"
+  [DEV]="dev"
+  [TEST]="test")
+prefix=${prefix:+${prefix}_}
+
+corpus_dir=$data_dir/${prefix}corpus/
+if [ $stage -le 1 ]; then
+  echo "$0: Extract meta into $corpus_dir"
+  # Sanity check.
+  [ ! -f $gigaspeech_dir/GigaSpeech.json ] &&\
+    echo "$0: Please download $gigaspeech_dir/GigaSpeech.json!" && exit 1;
+  [ ! -d $gigaspeech_dir/audio ] &&\
+    echo "$0: Please download $gigaspeech_dir/audio!" && exit 1;
+
+  [ ! -d $corpus_dir ] && mkdir -p $corpus_dir
+
+  # Files to be created:
+  # wav.scp text segments utt2dur
+  python3 ../../toolkits/wenet/extract_meta.py \
+     $gigaspeech_dir/GigaSpeech.json $corpus_dir | exit 1;
```

Comment:
We should probably assume that users will run the gigaspeech scripts under the repo root directory? So something like
```bash
  python3 toolkits/wenet/extract_meta.py ...
```